pkg/controller/controller_instance.go

```diff
@@ -1071,6 +1128,30 @@ func (c *controller) updateServiceInstanceCondition(
 	return err
 }
 
+// prepareObservedGeneration sets the instance's observed generation
+// and clears the conditions, preparing it for any status updates that can occur
+// during the further processing.
+// It doesn't send the update request to server.
+func (c *controller) prepareObservedGeneration(toUpdate *v1beta1.ServiceInstance) {
+	toUpdate.Status.ObservedGeneration = toUpdate.Generation
+	reason := observedNewGenerationReason
+	message := observedNewGenerationMessage
+	setServiceInstanceCondition(
```

Comment:
This condition will never actually surface to the user as far as I'm aware. Whenever we update the status of our resource, we also update the `Ready` condition, so it'll always be overwritten.